src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -848,10 +848,12 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 
 						var parameters = calledMethod.Parameters;
 						BindingFlags bindingFlags = BindingFlags.Public | BindingFlags.Instance;
-						if (parameters.Count > 1) {
-							if (methodParams[1].AsConstInt () != null)
-								bindingFlags |= (BindingFlags) methodParams[1].AsConstInt ();
-						}
+						bool unknownBindingFlags = false;
+						if (parameters.Count > 1 && calledMethod.Parameters[0].ParameterType.Name == "BindingFlags" && methodParams[1].AsConstInt () != null)
+							bindingFlags |= (BindingFlags) methodParams[1].AsConstInt ();
+						else if (parameters.Count > 1 && calledMethod.Parameters[0].ParameterType.Name == "BindingFlags" && methodParams[1].AsConstInt () == null)
+							unknownBindingFlags = true;
+
```

Comment:
I think I would probably prefer some kind of if/switch over all the overloads - and specifically set binding flags in those cases where they are specified (And maybe have a helper which checks the constant value of that parameter). And then specifically set them to the right defaults for overloads which don't specify them.

We could also use nullable to avoid the extra boolean flag.

Reply:
I think I got everything solved except for the part of an if/switch that goes over all the overloads, I think that asking for `ParameterType.Name == "BindingFlags"` is a good optimized way to solve if we are dealing with BindingFlags and doesn't require to go over all overloads.

Reply:
The overloads which do have `BindingFlags` parameter are the easy ones (Since they explicitly state binding flags). I'm more worried about the overloads which don't take the parameter. I guess we rely on the fact that those all have the same behavior - which might be OK. In that case I think we should at the very least add a comment about that. To make it clear it was a conscious decision. This is very much related to https://github.com/mono/linker/issues/1623 - that's why I'm very picky about this topic :-)

Reply:
I added a comment to reflect that we are using a default value for methods that don't use BindingFlags